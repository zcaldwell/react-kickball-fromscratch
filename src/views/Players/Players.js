import React from 'react';
import { useState, useEffect } from 'react';
import { fetchPlayers } from '../../services/players';
import PlayerList from '../../components/PlayerList';

export default function Players() {
  const [players, setPlayers] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchPlayers();
      setPlayers(data);
    };
    fetchData();
  }, []);

  return (
    <div>
      <PlayerList players={players} />
    </div>
  );
}
