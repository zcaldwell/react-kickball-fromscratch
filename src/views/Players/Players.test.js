import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Players from './Players';

test('Players should display all players', async () => {
  const { container } = render(
    <MemoryRouter>
      <Players />
    </MemoryRouter>
  );
  await screen.findByText('Bennie Jetts');
  expect(container).toMatchSnapshot();
});
