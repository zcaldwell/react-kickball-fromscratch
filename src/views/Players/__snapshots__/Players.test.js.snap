// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players should display all players 1`] = `
<div>
  <div>
    <div>
      <ul>
        <li>
          <a
            href="/players/1"
          >
            Bennie Jetts
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/12"
          >
            Betty Grey
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/7"
          >
            Bud E. Guy
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/8"
          >
            Buster Rimes
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/9"
          >
            Deb E. Downer
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/13"
          >
            Dwayne Elizondo Mountain Dew Herbert Camacho
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/15"
          >
            Frito P.
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/5"
          >
            Gary Coolman
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/2"
          >
            Hans Olo
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/10"
          >
            Jayne Phonda
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/14"
          >
            Joe Bauers
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/11"
          >
            Kara Whollot
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/3"
          >
            Ron Long
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/4"
          >
            Shorty Biggum
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/players/6"
          >
            Terry Cruise
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
